Userland/Libraries/LibPDF/Parser.cpp

```diff
@@ -113,21 +119,25 @@ bool Parser::parse_header()
     return true;
 }
 
-bool Parser::initialize_linearization_dict()
+Parser::LinearizationResult Parser::initialize_linearization_dict()
 {
     // parse_header() is called immediately before this, so we are at the right location
     auto dict_value = m_document->resolve(parse_indirect_value());
     if (!dict_value || !dict_value.is_object())
-        return false;
+        return LinearizationResult::NotLinearized;
 
     auto dict_object = dict_value.as_object();
     if (!dict_object->is_dict())
-        return false;
+        return LinearizationResult::NotLinearized;
 
     auto dict = object_cast<DictObject>(dict_object);
 
+    if (!dict->contains(CommonNames::Linearized)) {
+        return LinearizationResult::NotLinearized;
+    }
```

Comment:
Don't need the braces here either, but I do like this change. We now know that this is a linearization dictionary, so the below conditions are errors instead of just non-linearized. Nice one :)